image/image.go

```diff
@@ -293,6 +293,37 @@ func MockTrusted(mockTrusted []asserts.Assertion) (restore func()) {
 	}
 }
 
+func makeKernelChannel(kernelTrack, defaultChannel string) (string, error) {
+	kch, err := snap.ParseChannel(kernelTrack, "")
+	if err != nil {
+		return "", fmt.Errorf("cannot use kernel-track %q: %v", kernelTrack, err)
+	}
+	if defaultChannel != "" {
+		dch, err := snap.ParseChannel(defaultChannel, "")
+		if err != nil {
+			return "", fmt.Errorf("internal error: cannot parse channel %q", defaultChannel)
+		}
+		kch.Risk = dch.Risk
+	}
+	return kch.Clean().String(), nil
+}
+
+func amendKernelTrack(model *asserts.Model, defaultChannel string, local *localInfos) error {
+	if model.Kernel() == "" || model.KernelTrack() == "" {
+		return nil
+	}
+	info := local.Info(model.Kernel())
+	if info == nil {
+		return nil
+	}
+	ch, err := makeKernelChannel(model.KernelTrack(), defaultChannel)
+	if err != nil {
+		return err
+	}
+	info.Channel = ch
```

Comment:
This is the only side effect in this function, and it's unclear from the function prototype that this is the actual parameter mutated. One way to address it might be to have the function as a private method on `localInfos`.